theano/scan_module/scan_op.py

```diff
@@ -1607,14 +1607,18 @@ def infer_shape(self, node, input_shapes):
         inner_ins_shapes = []
         out_equivalent = OrderedDict()
 
+        # The two following blocks are commented as it cause in some
+        # cases extra scans in the graph. See gh-XXX for the
+        # investigation.
+
         # We skip the first outer input as it is the total or current number
         # of iterations.
         # sequences
-        seqs_shape = [x[1:] for x in input_shapes[1:1 + self.n_seqs]]
-        inner_seqs = self.inputs[:self.n_seqs]
-        outer_seqs = node.inputs[1:1 + self.n_seqs]
-        for in_s, out_s in izip(inner_seqs, outer_seqs):
-            out_equivalent[in_s] = out_s[0]
+        # seqs_shape = [x[1:] for x in input_shapes[1:1 + self.n_seqs]]
+        # inner_seqs = self.inputs[:self.n_seqs]
+        # outer_seqs = node.inputs[1:1 + self.n_seqs]
```

Comment:
These variables are still used below, so maybe there is more to comment out, or we should not comment out all of that.